plugin-server/src/worker/rusty-hook.ts

```diff
@@ -0,0 +1,114 @@
+import { Webhook } from '@posthog/plugin-scaffold'
+import * as Sentry from '@sentry/node'
+import fetch from 'node-fetch'
+import { isProdEnv } from 'utils/env-utils'
+import { raiseIfUserProvidedUrlUnsafe } from 'utils/fetch'
+
+import { ValueMatcher } from '../types'
+import { status } from '../utils/status'
+import { sleep } from '../utils/utils'
+import { pluginActionMsSummary } from './metrics'
+
+const RUSTY_HOOK_BASE_DELAY_MS = 100
+const MAX_RUSTY_HOOK_DELAY_MS = 30_000
+
+interface RustyWebhookPayload {
+    parameters: Webhook
+    metadata: {
+        team_id: number
+        plugin_id: number
+        plugin_config_id: number
+    }
+}
+
+export class RustyHook {
+    constructor(
+        private enabledForTeams: ValueMatcher<number>,
+        private serviceUrl: string,
+        private requestTimeoutMs: number
+    ) {}
+
+    public async enqueueIfEnabledForTeam({
```

Comment:
This is a mouthful, but once we migrate everyone over it will just become `enqueue` and return `void`.